src/SqlPersistence/SynchronizedStorage/StorageAdapter.cs

```diff
@@ -23,19 +31,29 @@ public Task<CompletableSynchronizedStorageSession> TryAdapt(OutboxTransaction tr
             CompletableSynchronizedStorageSession session = new StorageSession(outboxTransaction.Connection, outboxTransaction.Transaction, false, infoCache);
             return Task.FromResult(session);
         }
-        return EmptyResult;
+        return EmptyResultTask;
     }
 
-    public Task<CompletableSynchronizedStorageSession> TryAdapt(TransportTransaction transportTransaction, ContextBag context)
+    public async Task<CompletableSynchronizedStorageSession> TryAdapt(TransportTransaction transportTransaction, ContextBag context)
     {
         SqlConnection existingSqlConnection;
         SqlTransaction existingSqlTransaction;
         //SQL server transport in native TX mode
         if (transportTransaction.TryGet(out existingSqlConnection) && transportTransaction.TryGet(out existingSqlTransaction))
         {
             CompletableSynchronizedStorageSession session = new StorageSession(existingSqlConnection, existingSqlTransaction, false, infoCache);
-            return Task.FromResult(session);
+            return session;
         }
+
+        Transaction existingTransaction;
+        // Transport supports DTC and uses TxScope owned by the transport
+        if (transportTransaction.TryGet(out existingTransaction))
+        {
+            var connection = await connectionBuilder.OpenConnection();
+            CompletableSynchronizedStorageSession session = new StorageSession(connection, NoSqlTransaction, true, infoCache);
```

Comment:
@SzymonPobiega not a bad idea. I discussed today a few other scenarios with @SimonCropp I will run through those with you in my morning.